### PR TITLE
Added Crypto::Bcrypt + Crypto:Subtle

### PR DESCRIPTION
**Pull request by akaufmann**
Sorry, I killed the history with my rebase. It was a long and stressful day .... :( Second attempt!

### CODE REVIEW

**Comment by asterite** [comment]
No need to stress when coding in Crystal, it should be fun :-)

Did you copy the API from somewhere? I don't know much about bcrypt so if you can provide something I can test against it would be great (like, in Ruby or Go if I use these methods I get these results, and they are the same in this Crystal implementation).

Then, instead of internally using a Hash (`p = {} of String => String`) I would use a record or a class. But we can optimize that later if we have the specs :-)

File: src/crypto/bcrypt.cr [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+  MIN_COST = 4
+  MAX_COST = 31
+  DEFAULT_COST = 10
+  ENCODED_SALT_SIZE = 22
+  MIN_HASH_SIZE = 59
+  MAJOR_VERSION = '2'
+  MINOR_VERSION = 'a'
+
+  def digest(pass, cost = 10)
```
Comment (jhass) [inline src/crypto/bcrypt.cr]: You already extracted the magic value to a constant, why not use it here?

Comment (akaufmann) [inline src/crypto/bcrypt.cr]: Yep, good catch. We should use the constant.

**Comment by akaufmann** [comment]
The API is inspired by Lua's Bcrypt.

https://github.com/mikejsavage/lua-bcrypt

**Comment by ysbaddaden** [comment]
Sanity check: it's possible to crypt 1000 random words with bcrypt/ruby (or another implementation) and to verify them against bcrypt/crystal, and vice versa.

**Comment by asterite** [comment]
@akaufmann Thanks! We finally found some time to review this. If you run the spec with `--release` a spec fails, and it's because you are using `v :: UInt8` in some place. Note that that doesn't initialize `v` to zero, but to garbage.

**Pull request merged**